root/static/scripts/edit/components/ExternalLinkAttributeDialog.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2021 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+import mutate from 'mutate-cow';
+import ButtonPopover from '../../common/components/ButtonPopover';
+import type {LinkRelationshipT} from '../externalLinks';
+import DateRangeFieldset, {
+  partialDateFromField,
+  runReducer as runDateRangeFieldsetReducer,
+  type ActionT as DateRangeFieldsetActionT,
+} from './DateRangeFieldset';
+import {
+  createCompoundField,
+  createField,
+} from '../../edit/utility/createField';
+import {l} from '../../common/i18n';
+import {
+  applyAllPendingErrors,
+  hasSubfieldErrors,
+} from '../../../../utility/subfieldErrors';
+import {copyDatePeriodField} from '../utility/copyFieldData';
+
+type PropsT = {
+  onConfirm: (DatePeriodRoleT) => void,
+  relationship: LinkRelationshipT,
+};
+
+type StateT = {
+  +datePeriodField: DatePeriodFieldT,
+};
+
+type WritableStateT = {
+  ...StateT,
+  datePeriodField: WritableDatePeriodFieldT,
+};
+
+type ActionT =
+| {
+  +action: DateRangeFieldsetActionT,
+  +type: 'update-date-period',
+  }
+| {+type: 'reset'}
+| {+type: 'show-all-pending-errors'};
+
+const createInitialState = (props: PropsT): StateT => {
+  const relationship = props.relationship;
+  const beginDate = relationship.begin_date;
+  const endDate = relationship.end_date;
+
+  return {
+    datePeriodField: {
+      errors: [],
+      has_errors: false,
+      field: {
+        begin_date: createCompoundField(
+          'period.begin_date',
+          {
+            day: beginDate?.day ?? null,
+            month: beginDate?.month ?? null,
+            year: beginDate?.year ?? null,
+          },
+        ),
+        end_date: createCompoundField(
+          'period.end_date',
+          {
+            day: endDate?.day ?? null,
+            month: endDate?.month ?? null,
+            year: endDate?.year ?? null,
+          },
+        ),
+        ended: createField('period.ended', relationship.ended),
+      },
+      html_name: '',
+      id: 0,
+      type: 'compound_field',
+    },
+  };
+};
+
+const ExternalLinkAttributeDialog = (props: PropsT): React.MixedElement => {
+  const buttonRef = React.useRef<HTMLButtonElement | null>(null);
+  const [open, setOpen] = React.useState(false);
+  const initialState = createInitialState(props);
```

Comment:
can remove this once the relationship is stored in state, since you can access it from 'reset' that way
```suggestion
```